test/type/mutate.ts

```diff
@@ -62,9 +62,11 @@ export function useMutatorTypes() {
 
   mutate(async () => '1')
 
+  // @ts-expect-error
   mutate(async () => 1)
 
-  mutate(async () => 1, { populateCache: false })
+  // FIXME: this should work.
+  // mutate(async () => 1, { populateCache: false })
```

Comment:
@promer94 thanks for adding the test, maybe we should create an issue to follow up on this? it feels like a regression for `mutate()` API types

Reply:
The original pr #2708 did not fix this correctly so there is no regression here.

I also think we should not support this because the type implementation would be really complicated and not useful in practice. You should always make sure you data has same structure as cache when using the bounded mutate fn.

I just keep it here as a reference.

Reply:
right the type is number, I updated them in https://github.com/vercel/swr/pull/2781